### ISSUE TITLE
Reduce classes needed for a transform request

### ISSUE THREAD

**Issue by BenGalewsky** [ssl-hep/ServiceX_frontend#330]
_As an Analyzer I want to be able to submit a transform request with minimal boilerplate code and imports so I can cleanly and easily create a fresh transform request_

# Description
Remove as many classes and enums as possible.

Example
```python
sx = ServiceXClient(backend="testing4")

ds = sx.func_adl_dataset(["root://eospublic.cern.ch//eos/opendata/atlas/OutreachDatasets/mc_345060.ggH125_ZZ4lep.4lep.root",
                         codegen="uproot",
                         title="This is a simple example",
                         tree="mini")\
                   .Select(lambda e: {'lep_pt': e['lep_pt']}). \
                  Where(lambda e: e['lep_pt'] > 1000). \
                  as_pandas()
```

### Implications
1. We will interrogate the list of dataset IDs to determine if they are _rucio_ DIDs or explicit URIs.
2. The optional Rucio DID Finder settings will be provided as path arguments (`?maxFiles`) - this is already supported by the DID Finder
3. The result format will default to Root, but it can be overridden as an argument to `func_adl_dataset` - Use just strings for this :"parquet"

### Extra Credit
If the list of DIDs contains a list of Rucio DIDs then upgrade the dataset to be a DatasetGroup. One dataset for each DID

**Comment by alexander-held** [comment]
Thanks for turning this into a story! In the example above, I am not quite sure about the role of `dataset_id`: it is not being used. In my "ideal" zero-to-ServiceX example this `FileListDataset` would not be user-facing (at least not mandatory) and a list of strings is accepted directly in `func_adl_dataset` or equivalent.

One more thought about 2.: it would feel more natural to me to have some kind of `max_files` setting somewhere other than in the file path string. That implies losing the ability to easily set this per sample, but that might be good enough for debugging. Maybe it could be a keyword argument in `func_adl_dataset`?

A second "extra credit" idea would be a way to support groups in a very similar interface. This could be by accepting dicts instead of lists as the first `func_adl_dataset` argument (dict keys being the dataset group names, values being the paths), maybe there is another convenient way too.

**Comment by BenGalewsky** [comment]
> Thanks for turning this into a story! In the example above, I am not quite sure about the role of `dataset_id`: it is not being used. In my "ideal" zero-to-ServiceX example this `FileListDataset` would not be user-facing (at least not mandatory) and a list of strings is accepted directly in `func_adl_dataset` or equivalent.

Oops- forgot to delete that from the original I copied from. It's now gone

**Comment by NJManganelli** [comment]
The maxfiles, if you're willing to support it, could be of type: ```None | int | list[int] | dict[dataset_string, int]``` where the latter match the # of datasets (although yet another bridge-too-far-? could be a default value in the dict)
I agree that enabling the above story would be great from a user perspective!

**Comment by BenGalewsky** [comment]
Closed via #340
